gneiss/sort.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import pandas as pd
+from functools import partial
+from gneiss.util import match
+
+
+def mean_niche_estimator(abundances, gradient):
+    """ Estimates the mean niche of an organism.
+
+    Calculates the mean niche or an organism along a gradient.
+    This is done by calculating the expected value of an organism
+    across the gradient.
+
+    Parameters
+    ----------
+    abundances : pd.Series, np.float
+        Vector of fraction abundances of an organism over
+        a list of samples.
+    gradient : pd.Series, np.float
+        Vector of numerical gradient values.
+
+    Returns
+    -------
+    np.float :
+        The mean gradient that the organism lives in.
+    """
+    if len(abundances) != len(gradient):
```

Comment:
I don't feel strongly about this but normally we do something like this to minimize how many times you calculate the lenght of the iterables:
``` python
len_abundances = len(abundances)
len_gradient = len(gradient)
if len_abundances != len_gradient:
    ....
```